st2actions/st2actions/runners/ssh/paramiko_ssh.py

```diff
@@ -478,14 +479,14 @@ def _get_decoded_data(self, data):
             self.logger.exception('Non UTF-8 character found in data: %s', data)
             raise
 
-    def _get_pkey_object(self, key_material):
+    def _get_pkey_object(self, key_material, passphrase):
         """
         Try to detect private key type and return paramiko.PKey object.
         """
 
         for cls in [paramiko.RSAKey, paramiko.DSSKey, paramiko.ECDSAKey]:
             try:
-                key = cls.from_private_key(StringIO(key_material))
+                key = cls.from_private_key(StringIO(key_material), password=passphrase)
             except paramiko.ssh_exception.SSHException:
```

Comment:
I wonder what kind of exception is thrown on invalid passphrase... Is it the same one with different message or?